prov/sockets/src/sock_ep_msg.c

```diff
@@ -1111,6 +1111,8 @@ static void *sock_pep_listener_thread(void *data)
 
 			cm_entry->fid = &pep->pep.fid;
 			cm_entry->info = sock_ep_msg_process_info(conn_req);
+			if (!cm_entry)
```

Comment:
This check is coming too late.  We've already dereferenced the cm_entry pointer above it.